### ISSUE TITLE
Explanation needed 

### ISSUE THREAD

**Issue by tortoisecatfish** [chegewara/esp32s2-tinyusb-arduino-as-component#2]
Could you please elaborate more on this:

> To use it its required to pull arduino-esp32 repository branch esp32s2 to components folder and https://github.com/hathach/tinyusb repository t library folder: https://github.com/chegewara/esp32s2-tinyusb-arduino-as-component/tree/master/components/EspTinyUSB

Do you mean that it is required to pull the `arduino-esp32` repository (checkout branch esp32s2 after cloning) to the components folder of esp32s2-tinyusb-arduino-as-component and update the tinyusb inside [this](https://github.com/chegewara/esp32s2-tinyusb-arduino-as-component/tree/master/components/EspTinyUSB)?

**Comment by chegewara** [comment]
As you can see there is folder tinyusb in 
https://github.com/chegewara/esp32s2-tinyusb-arduino-as-component/tree/master/components/EspTinyUSB

It should be submodule from https://github.com/hathach/tinyusb, just something went wrong when i did prepare it.

**Comment by chegewara** [comment]
How about using this:
https://github.com/chegewara/esp-idf/tree/master/examples/peripherals/usb

**Comment by tortoisecatfish** [comment]
Thank you for your reply but could you please simply confirm what I asked?

**Comment by chegewara** [comment]
I did, here:


> As you can see there is folder tinyusb in
> https://github.com/chegewara/esp32s2-tinyusb-arduino-as-component/tree/master/components/EspTinyUSB
> 
> It should be submodule from https://github.com/hathach/tinyusb, just something went wrong when i did prepare it.

Is it not clear?

**Comment by tortoisecatfish** [comment]
> Do you mean that it is required to pull the `arduino-esp32` repository (checkout branch esp32s2 after cloning) to the components folder of esp32s2-tinyusb-arduino-as-component and update the tinyusb inside [this](https://github.com/chegewara/esp32s2-tinyusb-arduino-as-component/tree/master/components/EspTinyUSB)?

No not really because I asked about `arduino-esp32` as well in my initial question which is left unanswered. 

> Is it required to pull the `arduino-esp32` repository (checkout branch esp32s2 after cloning) to the components folder of esp32s2-tinyusb-arduino-as-component? 

Could you please answer this?

> Update the tinyusb inside [this](https://github.com/chegewara/esp32s2-tinyusb-arduino-as-component/tree/master/components/EspTinyUSB)?

I think you answered this question. Thank you for that.

**Comment by chegewara** [comment]
> Is it required to pull the `arduino-esp32` repository (checkout branch esp32s2 after cloning) to the components folder of esp32s2-tinyusb-arduino-as-component?

Yes, just bear in mind it was 7 months ago. Since then a lot changed, especially in esp-idf. Currently i am using `idf-realease/v4.2 with specific esp-idf commit, because most recent changes in esp-idf are not compatible with any arduino release/commit.
